src/Hasher/Md5Hasher.php

```diff
@@ -12,9 +12,8 @@ class Md5Hasher implements HasherInterface
 {
     public function hash($string)
     {
-        return substr(md5($string), 0, 8); // 8 hexits = 32bit
-
-        // 4 bytes of binary md5 data could also be used, but
-        // performance seems to be the same.
+        // 8 hexits = 32bit, which also allows us to forego having to check whether it's over PHP_INT_MAX.
+        // The substring is converted to an int since hex strings sometimes get treated as ints if all digits are ints and this results in unexpected sorting order.
+        return hexdoc(substr(md5($string), 0, 8));
```

Comment:
You will need to use `hexdec` here.

Right now this is dropping a fatal into the unit tests.